ctrd/image/image.go

```diff
@@ -0,0 +1,314 @@
+package image
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"net/http"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/platforms"
+	"github.com/containerd/containerd/remotes"
+	"github.com/containerd/containerd/remotes/docker"
+	"github.com/containerd/containerd/rootfs"
+	"github.com/containerd/containerd/snapshots"
+	"github.com/opencontainers/go-digest"
+	"github.com/opencontainers/image-spec/identity"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func defaultRemoteContext() *containerd.RemoteContext {
+	return &containerd.RemoteContext{
+		Resolver: docker.NewResolver(docker.ResolverOptions{
+			Client: http.DefaultClient,
+		}),
+		Snapshotter: containerd.DefaultSnapshotter,
+	}
+}
+
+// Pull downloads the provided content into containerd's content store
+func Pull(c *containerd.Client, ctx context.Context, ref string, includeLayer bool, opts ...containerd.RemoteOpt) error {
```

Comment:
How about put `context` to the first place?

Reply:
has updated